File: frontend/modules/identity/auth/types/response/SignInResponse.ts
import {MfaMethod} from "@identity/user/types/model/MfaMethod.ts";
import {User} from "@identity/user/types/model/User.ts";

export interface SignInResponse {
    user?: User;
    accessToken?: string;
    mfaRequired?: boolean;
    mfaMethod?: MfaMethod;
    emailVerificationRequired?: boolean
}

File: backend/modules/identity/src/main/java/bitecode/modules/auth/auth/model/response/SignInResponse.java
package bitecode.modules.auth.auth.model.response;

import bitecode.modules._common.shared.identity.user.model.data.UserDetails;
import bitecode.modules.auth.auth.model.enums.MfaMethod;
import jakarta.annotation.Nullable;
import lombok.Builder;

@Builder
public record SignInResponse(
        @Nullable
        UserDetails user,
        @Nullable
        String accessToken,
        @Nullable
        Boolean mfaRequired,
        @Nullable
        MfaMethod mfaMethod,
        Boolean emailVerificationRequired
) {
}
